### DISCUSSION TITLE
DataGrid customizing DataGridRowGroupHeader

### DISCUSSION THREAD

**Discussion by KrzysztofDusko** [AvaloniaUI/Avalonia#8502] [Q&A] [answered]
There is any way to change text in DataGridRowGroupHeader  ? 
Thre is option group.IsItemCountVisible but how can I change eg. 
"1000 Items"-> "1 000 items"
or 
"1000 Items"-> "1 000 items, avarage is 123, sum is 456" etc. 

Maybe is way to show this in RowDetailsDataGrid ? But rows details seems to not work for row group header.. 

I also tried DataGrid_LoadingRowGroup event but DataGridRowGroupHeaderEventArgs.RowGroupHeader is read only :(

**Comment by KrzysztofDusko** [comment, marked as answer]
ok.. 

        e.RowGroupHeader.Template = new FuncControlTemplate((x, y) =>
            new TextBlock
            {
               Text = "TEST"
            }
        );

**Reply by hjrb** [reply]
that is only some out of context code fragment. Nobody can use this🤔

**Reply by zangetsudo** [reply]
how to use it ????

**Reply by tm-mat** [reply]
You use it in the OnLoadingRowGroup event handler. However, it replaces the entire template including the expand/collapse button, not just the text.

**Reply by zangetsudo** [reply]
any more samples？please
